plugins/woocommerce/tests/php/includes/class-wc-checkout-test.php

```diff
@@ -97,6 +97,34 @@ public function test_order_notes() {
 		$this->assertStringContainsString( 'This text should not save inside an anchor.', $content );
 	}
 
+	/**
+	 * @testdox the customer notes can have linebreaks.
+	 */
+	public function test_order_notes_linebreaks() {
+		$data = array(
+			'ship_to_different_address' => false,
+			'order_comments'            => 'A string
+			with linebreaks
+			in it.',
+			'payment_method'            => WC_Gateway_BACS::ID,
+		);
+
+		$errors = new WP_Error();
+
+		$this->sut->validate_posted_data( $data, $errors );
+		$result = $this->sut->create_order( $data );
+
+		$content = wc_get_template_html(
+			'order/order-details.php',
+			array(
+				'order_id'       => $result,
+				'show_downloads' => false,
+			)
+		);
+		// The \n are necessary because those characters actually exist in the template.
+		$this->assertStringContainsString( 'A string<br />with linebreaks<br />in it.', preg_replace('/[\t|\\n]+/', '', $content) );
```

Comment:
Is the `preg_replace` fine here? he template outputs a bunch of tabs and spaces so I think it's easier to reason with this test if we replace them, rather than trying to account for them.